python/ray/train/tests/test_batch_predictor.py

```diff
@@ -147,7 +147,8 @@ def test_automatic_enable_gpu_from_num_gpus_per_worker(shutdown_only):
     with pytest.raises(
         ValueError, match="DummyPredictor does not support GPU prediction"
     ):
-        _ = batch_predictor.predict(test_dataset, num_gpus_per_worker=1)
+        ds = batch_predictor.predict(test_dataset, num_gpus_per_worker=1)
```

Comment:
This is the major behavior change needed to call out - `batch_predictor.predict()` becomes lazy now, and not force the execution. Users need to call `batch_predictor.predict().fully_executed()` to force the prediction to actually run.

The pros is we can chain multiple predictors now for free - `batch_predictor2.predict(batch_predictor1.predict())`.

The cons is this will probably be a surprising behavior change for current users.

If the impact of pros is not big, we can change to force execution inside `batch_predictor.predict()`.

cc @ericl, @amogkam.

Reply:
Cool, let me make the change to force execution inside predictor.

Reply:
Updated to force execution inside batch predictor.